### ISSUE TITLE
low validation accuracy

### ISSUE THREAD

**Issue by pengjiang030** [pytorch/examples#377]
Hi, 

I trained a resnet50 on imagenet. The training precision increases as expected, but the validation accurary is low (less than 1% after 50 epochs). Here is the log after 52 epochs:

Epoch: [52][4820/5005]	Time 0.917 (0.937)	Data 0.484 (0.494)	Loss 0.0236 (0.2188)	Prec@1 100.000 (94.143)	Prec@5 100.000 (99.676)
Epoch: [52][4830/5005]	Time 0.787 (0.937)	Data 0.345 (0.494)	Loss 0.0137 (0.2187)	Prec@1 100.000 (94.155)	Prec@5 100.000 (99.677)
Epoch: [52][4840/5005]	Time 0.789 (0.937)	Data 0.398 (0.494)	Loss 0.0020 (0.2185)	Prec@1 100.000 (94.166)	Prec@5 100.000 (99.677)
Epoch: [52][4850/5005]	Time 0.797 (0.937)	Data 0.353 (0.494)	Loss 0.2654 (0.2199)	Prec@1 100.000 (94.137)	Prec@5 100.000 (99.657)
Epoch: [52][4860/5005]	Time 1.135 (0.937)	Data 0.588 (0.494)	Loss 0.0935 (0.2195)	Prec@1 100.000 (94.149)	Prec@5 100.000 (99.658)
Epoch: [52][4870/5005]	Time 0.818 (0.937)	Data 0.468 (0.494)	Loss 0.2058 (0.2198)	Prec@1 100.000 (94.155)	Prec@5 100.000 (99.655)
Epoch: [52][4880/5005]	Time 1.117 (0.937)	Data 0.485 (0.494)	Loss 0.0103 (0.2199)	Prec@1 100.000 (94.147)	Prec@5 100.000 (99.656)
Epoch: [52][4890/5005]	Time 1.175 (0.937)	Data 0.455 (0.494)	Loss 0.0062 (0.2195)	Prec@1 100.000 (94.159)	Prec@5 100.000 (99.656)
Epoch: [52][4900/5005]	Time 1.011 (0.937)	Data 0.500 (0.494)	Loss 0.8957 (0.2194)	Prec@1 100.000 (94.171)	Prec@5 100.000 (99.657)
Epoch: [52][4910/5005]	Time 0.819 (0.937)	Data 0.549 (0.494)	Loss 0.5857 (0.2191)	Prec@1 100.000 (94.183)	Prec@5 100.000 (99.658)
Epoch: [52][4920/5005]	Time 0.865 (0.937)	Data 0.544 (0.494)	Loss 0.0858 (0.2191)	Prec@1 100.000 (94.195)	Prec@5 100.000 (99.658)
Epoch: [52][4930/5005]	Time 0.785 (0.937)	Data 0.571 (0.494)	Loss 0.0113 (0.2190)	Prec@1 100.000 (94.195)	Prec@5 100.000 (99.659)
Epoch: [52][4940/5005]	Time 0.813 (0.937)	Data 0.582 (0.494)	Loss 0.0171 (0.2194)	Prec@1 100.000 (94.185)	Prec@5 100.000 (99.660)
Epoch: [52][4950/5005]	Time 0.834 (0.937)	Data 0.401 (0.494)	Loss 0.1196 (0.2191)	Prec@1 100.000 (94.197)	Prec@5 100.000 (99.660)
Epoch: [52][4960/5005]	Time 0.917 (0.937)	Data 0.419 (0.494)	Loss 1.0364 (0.2192)	Prec@1 93.750 (94.207)	Prec@5 100.000 (99.661)
Epoch: [52][4970/5005]	Time 1.162 (0.937)	Data 0.616 (0.494)	Loss 0.0327 (0.2193)	Prec@1 100.000 (94.216)	Prec@5 100.000 (99.662)
Epoch: [52][4980/5005]	Time 1.074 (0.937)	Data 0.445 (0.494)	Loss 0.0076 (0.2191)	Prec@1 100.000 (94.208)	Prec@5 100.000 (99.662)
Epoch: [52][4990/5005]	Time 0.984 (0.937)	Data 0.438 (0.494)	Loss 0.0054 (0.2189)	Prec@1 100.000 (94.218)	Prec@5 100.000 (99.663)
Epoch: [52][5000/5005]	Time 0.811 (0.937)	Data 0.441 (0.494)	Loss 0.0052 (0.2187)	Prec@1 100.000 (94.230)	Prec@5 100.000 (99.664)
Test: [0/196]	Time 8.068 (8.068)	Loss 7.7088 (7.7088)	Prec@1 0.000 (0.000)	Prec@5 0.000 (0.000)
Test: [10/196]	Time 8.806 (8.611)	Loss 6.1847 (6.0084)	Prec@1 0.000 (0.000)	Prec@5 0.000 (4.545)
Test: [20/196]	Time 9.949 (8.706)	Loss 7.5897 (6.2823)	Prec@1 0.000 (1.079)	Prec@5 0.000 (5.171)
Test: [30/196]	Time 7.854 (8.769)	Loss 9.2412 (6.6962)	Prec@1 0.000 (0.731)	Prec@5 0.000 (3.503)
Test: [40/196]	Time 7.649 (8.905)	Loss 8.6179 (7.1559)	Prec@1 0.000 (0.553)	Prec@5 0.000 (2.649)
Test: [50/196]	Time 7.637 (8.708)	Loss 9.9357 (7.5217)	Prec@1 0.000 (0.444)	Prec@5 0.000 (2.129)
Test: [60/196]	Time 8.328 (8.603)	Loss 6.9728 (7.5828)	Prec@1 0.000 (0.371)	Prec@5 0.000 (1.780)
Test: [70/196]	Time 7.150 (8.544)	Loss 7.7414 (7.6249)	Prec@1 0.000 (0.319)	Prec@5 0.000 (1.529)
Test: [80/196]	Time 10.380 (8.538)	Loss 8.8565 (7.6701)	Prec@1 0.000 (0.280)	Prec@5 0.000 (1.341)
Test: [90/196]	Time 7.407 (8.435)	Loss 8.7305 (7.7553)	Prec@1 0.000 (0.249)	Prec@5 0.000 (1.193)
Test: [100/196]	Time 7.090 (8.368)	Loss 8.2933 (7.8245)	Prec@1 0.000 (0.224)	Prec@5 0.000 (1.075)
Test: [110/196]	Time 7.822 (8.299)	Loss 8.2748 (7.8796)	Prec@1 0.000 (0.204)	Prec@5 0.000 (0.978)
Test: [120/196]	Time 7.491 (8.304)	Loss 7.3618 (7.9008)	Prec@1 0.000 (0.187)	Prec@5 0.000 (0.897)
Test: [130/196]	Time 7.720 (8.355)	Loss 7.9997 (7.9406)	Prec@1 0.000 (0.173)	Prec@5 0.000 (0.829)
Test: [140/196]	Time 8.128 (8.290)	Loss 8.7477 (7.9593)	Prec@1 0.000 (0.161)	Prec@5 0.000 (0.770)
Test: [150/196]	Time 6.998 (8.249)	Loss 7.8027 (7.9721)	Prec@1 0.000 (0.150)	Prec@5 0.000 (0.719)
Test: [160/196]	Time 7.576 (8.250)	Loss 8.2594 (7.9811)	Prec@1 0.000 (0.141)	Prec@5 0.000 (0.674)
Test: [170/196]	Time 9.694 (8.224)	Loss 8.6592 (8.0005)	Prec@1 0.000 (0.132)	Prec@5 0.000 (0.635)
Test: [180/196]	Time 7.719 (8.212)	Loss 8.8477 (8.0289)	Prec@1 0.000 (0.125)	Prec@5 0.000 (0.600)
Test: [190/196]	Time 7.950 (8.228)	Loss 8.4233 (8.0592)	Prec@1 0.000 (0.119)	Prec@5 0.000 (0.569)
 * Prec@1 0.116 Prec@5 0.556

You can see that the training precion is above 90%, but the validation precion is less than 1%...

The command I used is: 
python main.py -a resnet50 /the/path/to/imagenet/

What can possbily be the problem? Thanks!

**Comment by avinabsaha** [comment]
I am facing the same issue, could you tell me what is the error?

**Comment by zb1439** [comment]
Same issue here! Wondering what is the problem.

**Comment by pengjiang030** [comment]
Hello, I used a shuffle function I implemented myself, and there was some bug in it. After I switched back to the shuffle function provided in Pytorch, the problem was gone and the accuracy went up. Hope this help.

**Comment by codeprb** [comment]
I am having the same error in the 1st epoch itself for imagenet data. Could you please elaborate the code block @pengjiang030 which you modified?

**Comment by PoonKinWang** [comment]
> I am having the same error in the 1st epoch itself for imagenet data. Could you please elaborate the code block @pengjiang030 which you modified?

What is your dataset? ImageNet?
